cinema/views.py

```diff
@@ -56,3 +84,61 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            try:
+                date = datetime.strptime(date, "%Y-%m-%d").date()
+                queryset = queryset.filter(show_time__date=date)
+            except ValueError:
+                pass
+
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+
+        if self.action in "list":
+            queryset = (
```

Comment:
The condition `if self.action in 'list':` should be `if self.action == 'list':` to correctly check if the action is 'list'. Using `in` with a string can lead to unexpected behavior.